### PR TITLE
#51: Spark-commons should depend on Spark-commons-test only in tests

### LINKED ISSUE #51: Spark-commons should depend on Spark-commons-test only in tests
## Describe the bug
_Spark-commons_ should depend on _Spark-commons-test_ only in tests

## To Reproduce
Check `build.sbt`:
```
lazy val `sparkCommons` = (projectMatrix in file("spark-commons"))
  .settings(commonSettings: _*)
  .sparkRow(SparkVersionAxis(spark2), scalaVersions = Seq(scala211, scala212))
  .sparkRow(SparkVersionAxis(spark3), scalaVersions = Seq(scala212))
  .dependsOn(sparkCommonsTest)
```

### PR DESCRIPTION
**Pull request by benedeki**
* dependency only for test
* updated project version

Closes #51

### CODE REVIEW

**Review by dk1844 — APPROVED** [review]
LGTM (
 -  [x] read, 
 -  [x] checked out,
 -  [x] sbt-`compile`d, sbt-`test`-ran
)

---
_Perhaps unnecessary comment, but it may help realize some details:_

Since 
```scala
sparkCommons.dependsOn(sparkCommonsTest % "test")
```
 actually means
```scala
sparkCommons.dependsOn(sparkCommonsTest % "test->compile")
```
 (that is `sparkCommons`'s `test` configuration actually depends on `sparkCommonsTest`'s `compile` configuration - see [here](https://www.scala-sbt.org/1.x/docs/Multi-Project.html#Per-configuration+classpath+dependencies) for explanation how `->compile` is implied), I wondered if test config should not rely on another project's `test` config instead - expressed perhaps as
```scala
sparkCommons.dependsOn(sparkCommonsTest % "test->test")
```

Long story short, in some cases it may be preferred (e.g. with projects that have some `main` code and `test` code and we only want to reuse the `test` code elsewhere), but here it is not so, because `sparkCommonsTest` `main` code  (`compile` config) is actually the core functionality of the project (module).

So, I consider this setup correct. At least I have learned something 😄
